projects/laji/src/app/+observation/observation-data.service.ts

```diff
@@ -31,105 +21,55 @@ const overrideType = {
   providedIn: 'root'
 })
 export class ObservationDataService {
+  cacheCount$: Observable<IObservationData>;
+  lastQuery: string;
 
   constructor(
-    private graphQLService: GraphQLService,
-    private historyService: HistoryService,
     private searchQueryService: SearchQueryService,
-    private platformService: PlatformService
+    private platformService: PlatformService,
+    private warehouseService: WarehouseApi
   ) { }
 
   getData(query: WarehouseQueryInterface): Observable<IObservationData> {
     if (this.platformService.isServer) {
       return of(this.empty());
     }
+    const newQuery = JSON.stringify(query);
+
+    if (newQuery === this.lastQuery) {
```

Comment:
This will give false negatives, if the "order" of properties changes (`{"a":1,"b":2} !== {"b":2,"a":1}`). You could use for example the `deep-equal` package for checking the similarity.

Reply:
Done with deep-equal.